[MIND-4] 내정보 수정 기능 구현

TaskerJang opened the pull request:
# 📝 내정보 수정 기능 구현

## 📌 작업 내용
- 내정보 수정을 위한 DTO 클래스 추가
- User 엔티티에 프로필 수정 메서드 추가
- 내정보 수정 API 구현

## ⚙️ 기술적 변경사항
- ApiResponse.java: API 응답 형식 표준화
- UpdateUserRequest.java: 내정보 수정 요청 DTO
- User.java: updateProfile 메서드 추가
- UserController.java: 내정보 수정 API 엔드포인트 추가
- UserService.java: 내정보 수정 로직 추가

## 🧪 테스트
- [x] 내정보 수정 API 호출 시 정상적으로 데이터가 업데이트됨
- [x] 내정보 조회 시 수정된 정보가 정상적으로 반환됨

Commit:
[feat]: 내정보 수정을 위한 DTO 클래스 추가

Commit:
[feat]: User 엔티티에 updateProfile 메서드 추가

Commit:
[feat]: 내정보 수정 API 구현

duhyeon99 reviewed (approved):
고생하셨습니다

hwan2-99 commented on src/main/java/com/mindgoal/domain/user/dto/ApiResponse.java:
```diff
@@ -0,0 +1,16 @@
+
+    public static <T> ApiResponse<T> success(T data, String message) {
+        return new ApiResponse<>(200, message, data);
+    }
```
해당 reponse 객체를 어떻게 관리할지 회의하면 좋을것같습니다!

TaskerJang replied:
네, 좋은 의견 감사합니다! 응답 객체 관리 방안에 대해 생각을 해보았는데, 다음과 같은 형태로 구현하면 될까용?

```java
// HTTP 상태 코드와 메시지를 가진 enum
public enum ApiStatus {
    SUCCESS(200, "요청이 성공했습니다"),
    BAD_REQUEST(400, "잘못된 요청입니다"),
    UNAUTHORIZED(401, "인증이 필요합니다");
    // ...
}

@Getter
public class ApiResponse<T> {
    private final int status;
    private final String message;
    private final T data;
    private final LocalDateTime timestamp;
    
    public static <T> ApiResponse<T> of(ApiStatus status, T data) {
        return new ApiResponse<>(status.getCode(), status.getMessage(), data);
    }
}
```

hwan2-99 replied:
좋은것 같습니다!!

hwan2-99 commented on src/main/java/com/mindgoal/domain/user/service/UserService.java:
```diff
@@ -110,4 +111,12 @@ public User createKakaoUser(KakaoUserInfo userInfo) {
+        user.updateProfile(request.getName(), request.getProfileImage());
+        return userRepository.save(user);
+    }
+
```
해당메서드에 대한 테스트가 구현되면 좋을것같습니다!

TaskerJang replied:
네, 좋은 피드백 감사합니다! 테스트 코드를 추가하도록 하겠습니다.

hwan2-99 reviewed (commented):
고생하셨습니다!

Commit:
[feat]: BaseResponse 클래스 구현
 - BaseResponse 클래스 추가
 - 공통 응답 형식 정의
 - 성공/실패 응답 처리 메서드 구현
 - timestamp 필드 추가

Commit:
[feat]: BaseResponseStatus enum 구현
- HTTP 상태 코드와 응답 메시지를 정의하는 enum 클래스 구현
- 상태 코드별 표준 메시지 정의
- 성공/실패 여부 플래그 추가

Commit:
[feat]: UpdateUserRequest DTO 수정

Commit:
[refactor]: UserController에 BaseResponse 적용
- 응답 형식을 BaseResponse로 통일
- 성공/실패 응답 메시지 표준화
- 에러 처리 개선

Commit:
[test]: UserService 테스트 코드 추가
- 사용자 정보 업데이트 성공 케이스 테스트 추가
- 존재하지 않는 사용자 업데이트 실패 케이스 테스트 추가
- Mock 객체를 활용한 테스트 환경 구성

Commit:
[remove]: ApiResponse 클래스 제거

hwan2-99 reviewed (approved):
고생하셨습니다!!
